internal/pkg/api/router.go

```diff
@@ -5,24 +5,37 @@
 package api
 
 import (
+	"encoding/json"
+	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/elastic/fleet-server/v7/internal/pkg/config"
-	"github.com/elastic/fleet-server/v7/internal/pkg/limit"
-	"github.com/elastic/fleet-server/v7/internal/pkg/logger"
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/chi/v5/middleware"
 	"github.com/rs/zerolog"
 	"go.elastic.co/apm/module/apmchiv5/v2"
 	"go.elastic.co/apm/v2"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/fleet-server/v7/internal/pkg/config"
+	"github.com/elastic/fleet-server/v7/internal/pkg/limit"
+	"github.com/elastic/fleet-server/v7/internal/pkg/logger"
+	"github.com/elastic/fleet-server/v7/internal/pkg/policy"
 )
 
-func newRouter(cfg *config.ServerLimits, si ServerInterface, tracer *apm.Tracer) http.Handler {
+func newRouter(cfg *config.ServerLimits, si ServerInterface, tracer *apm.Tracer, sm policy.SelfMonitor) http.Handler {
 	r := chi.NewRouter()
 	r.Use(logger.Middleware) // Attach middlewares to router directly so the occur before any request parsing/validation
 	r.Use(middleware.Recoverer)
 	r.Use(Limiter(cfg).middleware)
+
+	// In managed mode, fleet-server is expected to be able to operate normally during startup,
+	// so the agent managing it can enroll. So attach the statusChecker middleware only on
+	// stand-alone mode.
+	if _, isStandAlone := sm.(*policy.StandAloneSelfMonitor); isStandAlone {
+		r.Use(statusChecker(sm))
+	}
```

Comment:
Not sure if I like to have this difference in behaviour, but I think we may need it, we don't have the same healthcheck needings when running as standalone and when running inside agent.

Reply:
I think our alternative is to create a health endpoint and have the platform direct traffic if to fleet-server once ES is healthy. What do you think?

Reply:
Yeah, once we have healthchecks in place, the platform won't send traffic to us when we cannot reach ES. Should I remove this?
I started adding this after [this comment](https://github.com/elastic/fleet-server/pull/2693#discussion_r1230244522) about returning 503s when the healthcheck fails. But I see now that maybe you were referring to the platform and not directly here?

Reply:
I am removing the middleware I added, and we will rely on readiness probes.